developer_manual/app/schema.rst

```diff
@@ -1,3 +1,141 @@
+
+Therefore we support three kinds of migration steps: simple, sql and schema.
+
+There are currently 3 types of migrations supported:
+
+- simple: this allows general migration steps to be executed - pretty similar to the `repair steps <https://doc.owncloud.org/api/classes/OCP.Migration.IRepairStep.html>`_
+- sql: this allows you to add a list of sql commands to be executed
+- schema: migration steps of this kind allow various schema migration operations.
+
+Starting with ownCloud 10 this is the preferred way to perform any kind of migrations and is enabled by default within core.
+Any app which wants to use this mechanism has to enable it in :file:`appinfo/info.xml` by adding
+
+.. code-block:: xml
+
+  <use-migrations>true</use-migrations>
+
+
+As soon as migrations are enabled any existing :file:`appinfo/database.xml` will be ignored on upgrade of the app.
+Instead all migration steps which have not yet been executed will be executed.
+
+It is still necessary to increment the apps version number to trigger the execution of steps.
+
+How to develop a migration
+==========================
+
+1. Enable migrations by adding the xml tag to :file:`appinfo/info.xml`
+
+.. code-block:: xml
+
+  <use-migrations>true</use-migrations>
+
+2. Create a migration step
+
+.. code-block:: bash
+
+  ./occ migrations:generate app-name {simple, sql, schema}
+
+A simple migration step skeleton looks like this:
+
+.. code-block:: php
+
+    <?php
+    namespace OCA\testing\Migrations;
+
+    use OCP\Migration\ISimpleMigration;
+    use OCP\Migration\IOutput;
+
+    /**
+     * Auto-generated migration step: Please modify to your needs!
+     */
+    class Version20170213125339 implements ISimpleMigration {
+
+        /**
+         * @param IOutput $out
+         */
+        public function run(IOutput $out) {
+            // auto-generated - please modify it to your needs
+        }
+    }
+
+A sql migration step skeleton looks like this:
+
+.. code-block:: php
+
+    <?php
+    namespace OCA\testing\Migrations;
+
+    use OCP\IDBConnection;
+    use OCP\Migration\ISqlMigration;
+
+    /**
+     * Auto-generated migration step: Please modify to your needs!
+     */
+    class Version20170213125430 implements ISqlMigration {
+
+        /**
+         * @param IDBConnection $connection
+         * @return array of sql statements
+         */
+        public function sql(IDBConnection $connection) {
+            // auto-generated - please modify it to your needs
+        }
+    }
+
+Within the sql() you can generate any number of sql commands which are returned as array. The statements are executed
+after the method returns. Please do not directly execute any statements on the database parameter.
+The parameter $connection can be used to e.g. get the database platform or to test if tables exist.
+
+In order to allow cross compatible sql code please use the platform object or generate sql commands for each supported
```

Comment:
I think you should capitalize SQL everywhere that is not a PHP symbol. Ex: "SQL commands"